associations/belongs_to_association.go

```diff
@@ -40,17 +50,40 @@ func belongsToAssociationBuilder(p associationParams) (Association, error) {
 	if fieldIsNil(f) || isZero(f.Interface()) {
 		skipped = true
 	}
+	//associated model
+	ownerPrimaryTableField := "id"
+	if primaryIDField != "ID" {
+		var ownerModel reflect.Value
```

Comment:
@u007 Just one thing I noticed, you should probably use [reflect.Indirect](https://golang.org/pkg/reflect/#Indirect) method instead. And you would save these conditionals statements.  Beside this I think this looks good.

Reply:
hi @larrymjordan , did u mean this line?
```
if fval.Kind() == reflect.Ptr {
	ownerModel = fval.Elem()
} else {
	ownerModel = fval
}

```

sorry im not really sure how indirect is used. With Indirect, does it means i dont have to check for pointer?

Reply:
@u007 That's correct, in this case,`Indirect` would return the value that `fval` is pointing to, if fval is a pointer.

Reply:
done :) pls check if its good